docs/reference/mount.md

```diff
@@ -52,6 +52,53 @@ Searching for lakeFS credentials and server endpoint in the following order:
 - `LAKECTL_*` Environment variables
 - `~/.lakectl.yaml` Configuration file or via `--lakectl-config` flag
 
+## Authenticating with AWS IAM Role
```

Comment:
Link to the IAM feature for people wanting to understand it better (https://docs.lakefs.io/security/external-principals-aws.html)